portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -3278,9 +3278,13 @@ private void _overwritePreviousFileVersion(
 
 		// File
 
+		DLFileVersion previousDLFileVersion =
+			_dlFileVersionLocalService.getLatestFileVersion(
+				dlFileEntry.getFileEntryId(), true);
+
```

Comment:
Isn't this the same thing as `latestDLFileVersion`? Also, shouldn't we delete it *after* trying to copy its contents?

Reply:
> Isn't this the same thing as latestDLFileVersion?

No. "latestDLFileVersion" in this context, is, in fact, the new dl file version that is been stored

In the past it worked because the filenames were the same (when the pattern was adding only the version, for example 1.0), so what the system did was removing this previous file (named as the new one).

Now the name of the new fileEntry has a UUID that is different in each iteration, so we have to get the PERSISTED dlfilentryVersion to get its name and remove the file.

> Also, shouldn't we delete it after trying to copy its contents?

I haven't change the order of deletion. I works as it did, but I'm getting the right name of the previous file system document.

Reply:
Maybe, as change, I could delete both (last and previous if their .getStoreFileName() is different) 

What do you think?